aiohttp/client_middlewares.py

```diff
@@ -0,0 +1,58 @@
+"""Client middleware support."""
+
+from collections.abc import Awaitable, Callable
+
+from .client_reqrep import ClientRequest, ClientResponse
+
+__all__ = ("ClientMiddlewareType", "ClientHandlerType", "build_client_middlewares")
+
+# Type alias for client request handlers - functions that process requests and return responses
+ClientHandlerType = Callable[..., Awaitable[ClientResponse]]
```

Comment:
Looks like this needs to be ClientRequest:
```suggestion
ClientHandlerType = Callable[[ClientRequest], Awaitable[ClientResponse]]
```